core/market/decentralized/tests/test_offer_broadcast.rs

```diff
@@ -0,0 +1,47 @@
+mod utils;
+
+#[cfg(test)]
+mod tests {
+    use crate::utils::MarketsNetwork;
+
+    use ya_client::model::market::Offer;
+    use ya_market_decentralized::MarketService;
+
+    use serde_json::json;
+    use std::sync::Arc;
+    use std::time::Duration;
+
+    /// Test adds offer. It should be broadcasted to other nodes in the network.
+    #[cfg_attr(not(feature = "market-test-suite"), ignore)]
+    #[actix_rt::test]
+    async fn test_broadcast_offer() -> Result<(), anyhow::Error> {
+        let network = MarketsNetwork::new("test_broadcast_offer")
+            .await
+            .add_market_instance("Node-1")
+            .await?
+            .add_market_instance("Node-2")
+            .await?
+            .add_market_instance("Node-3")
+            .await?;
+
+        // Add Offer on Node-1. It should be propagated to remaining nodes.
+        let market1: Arc<MarketService> = network.get_market("Node-1");
+        let identity1 = network.get_default_id("Node-1");
+
+        let offer = Offer::new(json!({}), "()".to_string());
+        let subscription_id = market1.subscribe_offer(offer, identity1.clone()).await?;
+
+        // Expect, that Offer will appear on other nodes.
+        let market2: Arc<MarketService> = network.get_market("Node-2");
+        let market3: Arc<MarketService> = network.get_market("Node-3");
+
+        // Wait for Offer propagation.
+        // TODO: How to wait without assuming any number of seconds?
+        tokio::time::delay_for(Duration::from_secs(1)).await;
+
+        assert!(market2.matcher.get_offer(&subscription_id).await?.is_some());
```

Comment:
pls try; seems to be failing not only because of ids, but also properties serde is wrong 
```suggestion
        assert_eq!(Some(offer), market2.matcher.get_offer(&subscription_id).await?);
```

Reply:
this also is failing
```suggestion
        let subscribed_offer = market1.matcher.get_offer(&subscription_id).await?;
        assert_eq!(subscribed_offer, market2.matcher.get_offer(&subscription_id).await?);
```

Reply:
Thanks. Very good catch. I didn't expected that something so obvious like serialization would fail me